src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java

```diff
@@ -298,13 +307,33 @@ static MemberName linkCallSiteTracing(Class<?> caller,
         Object bsmReference = bootstrapMethod.internalMemberName();
         if (bsmReference == null)  bsmReference = bootstrapMethod;
         String staticArglist = staticArglistForTrace(staticArguments);
-        System.out.println("linkCallSite "+caller.getName()+" "+
+        String callerName = caller.getName();
+        String[] callerInfo = new String[] {callerName};
+        StackWalker.getInstance().walk(new Function<Stream<StackWalker.StackFrame>, Object>() {
```

Comment:
You can call `StackWalker.getInstance(Option.RETAIN_CLASS_REFERENCE)` to retain `Class` instance in `StackFrame` such that you can compare `Class` instance rather than class name.

Reply:
The `StackWalker` API depends on Streams, which is likely to have lambdas in its implementation. To avoid the possibility of bootstrap problems, I have changed the code to use `Thread.currentThread().getStackTrace()`, which is used by `Throwable.printStackTrace()`. Since `printStackTrace()` is supposed to be useable at very early bootstrap stages, we are sure no lambdas will be used.

Unfortunately `getStackTrace()` doesn't return the `Class` object, so I have to use the class's name.